File: PART04/Calculator/android/app/src/main/java/com/calculator/CalculatorModule.kt
package com.calculator

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class CalculatorModule(reactContext: ReactApplicationContext): ReactContextBaseJavaModule(reactContext) {
    override fun getName() = "CalculatorModule" // module명을 바로 return ... js에서 사용 예정

    @ReactMethod
    fun executeCalc(action:String, numberA:Double, numberB:Double, promise: Promise) {
        if ( action === "plus" ) {
            promise.resolve(numberA + numberB)
            return;
        }
        if ( action === "minus" ) {
            promise.resolve(numberA - numberB)
            return;
        }
        if ( action === "multiply" ) {
            promise.resolve(numberA * numberB)
            return;
        }
        if ( action === "divide" ) {
            promise.resolve(numberA / numberB)
            return;
        }

        // 해당하는 것이 없는 경우
        promise.reject("Unexpected action type")
    }
}

File: PART04/Calculator/ios/CalculatorModule.swift
//
//  CalculatorModule.swift
//  Calculator
//
//  Created by isdiscodead on 10/6/23.
//

import Foundation

@objc(CalculatorModule)
class CalculatorModule:NSObject {
  // 실제로 돌아갈 함수
  @objc(executeCalc:numberA:numberB:resolver:rejector:)
  public func executeCalc(_ action: String, numberA: Int, numberB: Int, resolver:RCTPromiseResolveBlock, rejector: RCTPromiseRejectBlock) -> Void {
    
    if ( action == "plus" ) {
      resolver(numberA + numberB);
      return;
    }
    if ( action == "minus" ) {
      resolver(numberA - numberB);
      return;
    }
    if ( action == "multiply" ) {
      resolver(numberA * numberB);
      return;
    }
    if ( action == "divide" ) {
      resolver(numberA / numberB);
      return;
    }
    
    // 해당되지 않을 경우
    rejector("Unexpected action type", action, nil)
  }
  
}
